aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -0,0 +1,894 @@
+    
+        // Increment and retrieve the nonce  
+        let nonce = increment_and_get_nonce(initiator_address);  
+    
+        // Create bridge transfer details  
+        let details = native_bridge_store::create_details(  
+            initiator_address,  
+            ethereum_address, 
+            amount,  
+            nonce  
+        );  
+    
+        // Generate a unique bridge transfer ID  
+        // Todo: pass the nonce in here and modify the function to take a nonce. Or only use the nonce in native_bridge_store
+        let bridge_transfer_id = native_bridge_store::bridge_transfer_id(&details);  
+    
+        // Add the transfer details to storage  
+        native_bridge_store::add(bridge_transfer_id, details);
+    
+        // Push details to be able to lookup by nonce
+        native_bridge_store::set_nonce_to_bridge_transfer_id(nonce, bridge_transfer_id);
+
+        // Burn the amount from the initiator  
+        native_bridge_core::burn(initiator_address, amount);  
+    
+        let bridge_events = borrow_global_mut<BridgeEvents>(@aptos_framework);
+
+        // Emit an event with nonce  
+        event::emit_event(  
+             &mut bridge_events.bridge_transfer_initiated_events,
+            BridgeTransferInitiatedEvent {  
+                bridge_transfer_id,  
+                initiator: initiator_address,  
+                recipient,  
+                amount,  
+                nonce,  
+            }  
+        );  
+    }
+
+     /// Completes a bridge transfer by the initiator.  
+    ///  
+    /// @param caller The signer representing the bridge operator.  
+    /// @param initiator The initiator's Ethereum address as a vector of bytes.  
+    /// @param bridge_transfer_id The unique identifier for the bridge transfer.  
+    /// @param recipient The address of the recipient on the Aptos blockchain.  
+    /// @param amount The amount of assets to be locked.  
+    /// @param nonce The unique nonce for the transfer.    
+    /// @abort If the caller is not the bridge operator.  
+    public entry fun complete_bridge_transfer(  
+        caller: &signer,  
+        bridge_transfer_id: vector<u8>,
+        initiator: vector<u8>,  
+        recipient: address,  
+        amount: u64,  
+        nonce: u64  
+    ) acquires BridgeEvents {  
+        native_bridge_configuration::assert_is_caller_operator(caller);  
+        let ethereum_address = ethereum::ethereum_address_no_eip55(initiator);
+
+        let combined_bytes = vector::empty<u8>();
+        vector::append(&mut combined_bytes, bcs::to_bytes(&initiator));
+        vector::append(&mut combined_bytes, bcs::to_bytes(&recipient));
+        vector::append(&mut combined_bytes, bcs::to_bytes(&amount));
+        vector::append(&mut combined_bytes, bcs::to_bytes(&nonce));
+        assert!(keccak256(combined_bytes) == bridge_transfer_id, EINVALID_BRIDGE_TRANSFER_ID);
+        // todo: expect it to be empty
+ 
```

Comment:
You don't verify the transfer hasn't already been done. If the relayer can call several times complete for the same transfer.